cli/lib/nf3.mjs

```diff
@@ -928,18 +979,22 @@ class Nf3 {
       const { type, txDataToSign, block, transactions, data } = msg;
       logger.debug(`Proposer received websocket message of type ${type}`);
       if (type === 'block') {
+        // First sign transaction, and send it within asynchronous queue. This will
+        // ensure that blockProposed events are emitted in order and with the correct nonce.
+        const tx = await this._signTransaction(
```

Comment:
Maybe the _nonce_ itself can control the sequence of the blocks and we don't need to worry about this, I "guess".

Reply:
Not sure what `this` is. But we need to split the nonce computation and the transaction submission. The nonce computation is done serially, and then we don’t care about the transaction ordering because it comes given by the nonce